core/server/server_scion.go

```diff
@@ -281,13 +283,50 @@ func runSCIONServer(ctx context.Context, log *zap.Logger, mtrcs *scionServerMetr
 				continue
 			}
 
+			var NTSauthenticated bool = false
+			var ntsreq nts.NTSPacket
+			var serverCookie ntske.ServerCookie
+			if len(udpLayer.Payload) > ntp.PacketLen {
+				cookie, err := nts.ExtractCookie(udpLayer.Payload)
+				if err != nil {
+					log.Info("failed to extract cookie", zap.Error(err))
+					continue
+				}
+
+				var encryptedCookie ntske.EncryptedServerCookie
+				err = encryptedCookie.Decode(cookie)
+				if err != nil {
+					log.Info("failed to decode cookie", zap.Error(err))
+					continue
+				}
+
+				key, ok := provider.Get(int(encryptedCookie.ID))
+				if !ok {
+					log.Info("failed to get key", zap.Error(err))
+					continue
+				}
+
+				serverCookie, err = encryptedCookie.Decrypt(key.Value)
+				if err != nil {
+					log.Info("failed to decrypt cookie", zap.Error(err))
+					continue
+				}
+
+				err = nts.DecodePacket(&ntsreq, udpLayer.Payload, serverCookie.C2S)
+				if err != nil {
+					log.Info("failed to decode packet", zap.Error(err))
+					continue
+				}
+				NTSauthenticated = true
```

Comment:
Add `zap.Bool("ntsauth", ntsAuthenticated),` to the log statement "received request" below.

Reply:
Done